### PR TITLE
WebHost: noscript faq and glossary

### PR DESCRIPTION
**Pull request by Berserker66**
## What is this fixing or adding?
Makes /faq and /glossary work with noscript/uMatrix and similar, by not using JS anymore and rendering the md to html on server.

## How was this tested?
locally

## If this makes graphical changes, please attach screenshots.
I did not see a difference, which is the point.

### CODE REVIEW

File: WebHostLib/requirements.txt [hunk 1]
```diff
@@ -9,3 +9,4 @@ bokeh>=3.1.1; python_version <= '3.8'
 bokeh>=3.4.3; python_version == '3.9'
 bokeh>=3.5.2; python_version >= '3.10'
 markupsafe>=2.1.5
+Markdown>=3.7
```
Comment (black-sliver) [inline WebHostLib/requirements.txt:12]: missing newline at end of file

**Review by black-sliver — CHANGES_REQUESTED** [review]
The bullets in glossary -> world do not render correctly

File: WebHostLib/requirements.txt (hunk 1 above)
Comment (Berserker66) [inline WebHostLib/requirements.txt:12]:
```suggestion
Markdown>=3.7

```

**Comment by Berserker66** [comment]
Adding bugfix tag as the pages don't work for some users currently.

**Comment by black-sliver** [comment]
Hm, it now renders correctly, but there is still a slight difference in the DOM. I'm gonna have one more looksie

File: WebHostLib/misc.py [hunk 2]
```diff
@@ -69,14 +70,28 @@ def tutorial_landing():
...
-def faq(lang):
-    return render_template("faq.html", lang=lang)
+def faq(lang: str):
+    import markdown
+    with open(os.path.join(app.static_folder, "assets", "faq", secure_filename(lang)+".md")) as f:
+        document = f.read()
+    return render_template(
+        "markdown_document.html",
+        title="Frequently Asked Questions",
+        html_from_markdown=markdown.markdown(document,  extensions=["mdx_truly_sane_lists", "mdx_breakless_lists"]),
```
Comment (black-sliver) [inline WebHostLib/misc.py:80]:
```suggestion
        html_from_markdown=markdown.markdown(document,  extensions=["mdx_breakless_lists"]),
```

File: WebHostLib/misc.py [hunk 3]
```diff
@@ -69,14 +70,28 @@ def tutorial_landing():
...
-def terms(lang):
-    return render_template("glossary.html", lang=lang)
+def glossary(lang: str):
+    import markdown
+    with open(os.path.join(app.static_folder, "assets", "glossary", secure_filename(lang)+".md")) as f:
+        document = f.read()
+    return render_template(
+        "markdown_document.html",
+        title="Glossary",
+        html_from_markdown=markdown.markdown(document,  extensions=["mdx_truly_sane_lists", "mdx_breakless_lists"]),
```
Comment (black-sliver) [inline WebHostLib/misc.py:93]:
```suggestion
        html_from_markdown=markdown.markdown(document,  extensions=["mdx_breakless_lists"]),
```

File: WebHostLib/requirements.txt [hunk 4]
```diff
@@ -9,3 +9,6 @@ bokeh>=3.1.1; python_version <= '3.8'
...
+mdx_truly_sane_lists>=1.3
```
Comment (black-sliver) [inline WebHostLib/requirements.txt:13]:
```suggestion
```

**Review by black-sliver — CHANGES_REQUESTED** [review]
Sane lists are not sane after all.

With just `mdx_breakless_lists` it produces a DOM very similar to github and the old js code.

**Review by black-sliver — APPROVED** [review]
Link to the discord thread: <https://discord.com/channels/731205301247803413/1295539478454079498/1296211476826292304>

**Comment by LegendaryLinux** [comment]
Do these pages still auto-scroll to the `#target-div` if provided in the URL?

**Pull request merged**